core/psh/sysexec.c

```diff
@@ -25,15 +27,91 @@ void psh_sysexecinfo(void)
 }
 
 
+static int psh_sysexec_wldargmatch(const char *wld, int argc, char **argv, char const *wldend)
+{
+	int len = 0, ii;
+	const char *currc = wld, *nextc;
+
+	for (ii = 0; ii < argc; ii++) {
+
+		if ((nextc = strchr(currc, ' ')) == NULL || nextc > wldend)
+			nextc = wldend;
+		len = nextc - currc;
+		if (strlen(argv[ii]) != len || memcmp(argv[ii], currc, len) != 0)
+			return -1;
+
+		if (nextc == wldend || ii == (argc - 1))
+			break;
+
+		currc = nextc + 1;
+	}
+	
+	if (nextc == wldend && ii == (argc - 1))
+		return 1;
+	else
+		return -1;
+}
+
+
+static int psh_sysexec_checkcommand(int argc, char **argv)
+{
+	int acc = -1;
+	static const char wldlist[] = PSH_WILDCARDS;
+	const char *curr, *cmdend;
+	char wldbuff[80] = "";
+	FILE *wldfile;
+
+	/* check against /etc/wldcrds file */
+	if ((wldfile = fopen("/etc/wldcrds", "r")) != NULL) {
+		acc = 0;
+		while (fgets(wldbuff, sizeof(wldbuff), wldfile) != NULL && acc != 1) {
+			if (isprint(wldbuff[sizeof(wldbuff) - 2])) {
+				while (isprint(fgetc(wldfile)))
+				;
+				memset(wldbuff, 0, sizeof(wldbuff));
+				continue;
+			}
+
+			if ((cmdend = strchr(wldbuff, '\n')) == NULL)
+				cmdend = &wldbuff[strlen(wldbuff)];
```

Comment:
```suggestion
			cmdend = wldbuff + strcspn(wldbuff, "\n");
```
Maybe it would be simpler to use the `strcspn()` function.

Reply:
Yes! That was exactly what was needed here - thank you very much!